### PR TITLE
Feature/custom buttons remote

### PR DESCRIPTION
**Pull request by nikos1988**
PR 2/3

### CODE REVIEW

File: src/yio-plugin/integration.cpp [hunk 1]
```diff
@@ -124,3 +140,7 @@ void Integration::setState(int state) {
             break;
     }
 }
+
+void Integration::sendCustomCommand(const QString& type, const QString& entityId, int command, const QVariant& param) {
+    qWarning() << "send custom command not implemented";
```
Comment (zehnm) [inline src/yio-plugin/integration.cpp]: Please use the category logger: `qCWarning(m_logCategory) << ...`

File: src/yio-interface/integrationinterface.h [hunk 2]
```diff
@@ -53,6 +53,20 @@ class IntegrationInterface {
...
 
+    /**
+     * @brief addAvailableEntity
+     * @param entityId
+     * @param type
+     * @param integration
+     * @param friendlyName
+     * @param supportedFeatures
+     * @param customFeatures
+     */
```
Comment (zehnm) [inline src/yio-interface/integrationinterface.h]: Either document method or remove empty template.
Yes, there are other empty doc templates in this class which need to be fixed... But new code should be better ;-)

File: src/yio-interface/integrationinterface.h [hunk 3]
```diff
@@ -62,6 +76,15 @@ class IntegrationInterface {
      */
     virtual void sendCommand(const QString& type, const QString& entityId, int command, const QVariant& param) = 0;
 
+    /**
+     * @param type
+     * @param entityId
+     * @param command
+     * @param param
+     */
```
Comment (zehnm) [inline src/yio-interface/integrationinterface.h]: See above

File: src/yio-interface/integrationinterface.h [hunk 4]
```diff
@@ -62,6 +76,15 @@ class IntegrationInterface {
...
+    virtual void sendCustomCommand(const QString& type, const QString& entityId, int commandIndex,
```
Comment (zehnm) [inline src/yio-interface/integrationinterface.h]: Why is `sendCustomCommand` required what `sendCommand` can't handle?
Couldn't it be integrated in `sendCommand` with additional information in `param`?

**Review by zehnm — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Please add a pull request description, what this changes are for. Especially for the new `sendCustomCommand` method.

**Comment by nikos1988** [comment]
Hi, could you take another look at the two commits i added. I think supporting the custom features is possible with minimal changes to the addAvailableEntity method. (Added customFeatures string list parameter with empty list as default)

**Comment by zehnm** [comment]
@nikos1988 I'll have a look tomorrow. I didn't see the 3rd part of your PR, maybe I just missed it. Shouldn't there be an implementation in an integration plugin, e.g. home-assistant, openhab or homey?
I saw a mqtt reference in your other PR. What's that for, an upcoming mqtt integration?

**Comment by nikos1988** [comment]
> 
> @nikos1988 I'll have a look tomorrow. I didn't see the 3rd part of your PR, maybe I just missed it. Shouldn't there be an implementation in an integration plugin, e.g. home-assistant, openhab or homey?
> I saw a mqtt reference in your other PR. What's that for, an upcoming mqtt integration?

Hi, the 3rd PR was for the web-configurator to add the custom features. I have developed an mqtt integration, but this is currently customized for my own needs. Maybe i can release something better suited for everyone in the future.

**Comment by zehnm** [comment]
> I have developed an mqtt integration, but this is currently customized for my own needs.

Maybe we can join our efforts: I'm working on a webhook integration which would be very easy to extend with mqtt:
https://github.com/zehnm/integration.webhook
I've already thought of enhancing it with mqtt, since most logic could be used for webhook & mqtt, but haven't had time to continue working on it.

**Comment by zehnm** [comment]
Since there hasn't been any activity for a while and the PR looks ready, I'm going to change the state to ready and review it.

**Review by zehnm — APPROVED** [review]

**Pull request merged**